app/Http/Controllers/GistsController.php

```diff
@@ -25,15 +25,14 @@ public function __construct(GistlogRepository $repository)
     public function storeAndRedirect()
     {
         $gistUrl = Input::get('gistUrl');
-
         try {
             $gistlog = $this->repository->findByUrl($gistUrl);
         } catch (InvalidUrlException $e) {
             Session::flash('error-message', 'Please enter a valid Gist URL.');
-            return Redirect::route('home');
+            return Redirect::back();
```

Comment:
In the future, it's helpful to keep changes like this in separate PRs so they can be accepted and rejected separately; but good call on this. :)